example/test.py

```diff
@@ -77,25 +104,49 @@ def test_expected_processes(self, web_only_container):
             if len(ps_data) == 5:
                 break
             time.sleep(delay)
-            ps_data = list_container_processes(web_only_container)
+            ps_data = filter_ldconfig_process(
+                list_container_processes(web_only_container))
 
-        assert_tini_pid_1(ps_data.pop(0), 'mysite.wsgi:application')
+        assert_that(ps_data, HasLength(5))
 
-        ps_data = filter_ldconfig_process(ps_data)
+        tini = ps_data[0]
+        assert_tini_pid_1(tini, 'mysite.wsgi:application')
 
-        # The next processes we have no control over the start order or PIDs...
-        assert_that(ps_data, matches_ruser_args_unordered(
-            ('root', 'nginx: master process nginx -g daemon off;'),
-            ('nginx', 'nginx: worker process'),
-            ('django',
+        gunicorn_rows = [row for row in ps_data if 'gunicorn' in row.args]
+        gunicorn_master, gunicorn_workers = (
+            find_prefork_worker_split(gunicorn_rows))
+        assert_that(gunicorn_master, matches_attributes_values(
+            ('ppid', 'ruser', 'args'),
+            (tini.pid, 'django',
              '/usr/local/bin/python /usr/local/bin/gunicorn '
              'mysite.wsgi:application --pid /var/run/gunicorn/gunicorn.pid '
-             '--bind unix:/var/run/gunicorn/gunicorn.sock --umask 0117'),
-            # No obvious way to differentiate Gunicorn master from worker
-            ('django',
+             '--bind unix:/var/run/gunicorn/gunicorn.sock --umask 0117')
+        ))
+
+        assert_that(gunicorn_workers, HasLength(1))
+        [gunicorn_worker] = gunicorn_workers
+        assert_that(gunicorn_worker, matches_attributes_values(
+            ('ppid', 'ruser', 'args'),
+            (gunicorn_master.pid, 'django',
              '/usr/local/bin/python /usr/local/bin/gunicorn '
              'mysite.wsgi:application --pid /var/run/gunicorn/gunicorn.pid '
-             '--bind unix:/var/run/gunicorn/gunicorn.sock --umask 0117'),
+             '--bind unix:/var/run/gunicorn/gunicorn.sock --umask 0117')
+        ))
+
+        nginx_rows = [row for row in ps_data if 'nginx' in row.args]
+        nginx_master, nginx_workers = find_prefork_worker_split(nginx_rows)
+        assert_that(nginx_master, matches_attributes_values(
+            ('ppid', 'ruser', 'args'),
+            # FIXME: Nginx should not be parented by Gunicorn
```

Comment:
I'm guessing this is because we `exec` the `gunicorn` process at the end of our bootstrap script? In that case, everything we run in there will be parented by it.